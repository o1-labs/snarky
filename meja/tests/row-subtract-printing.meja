let refine = fun (x : 'a,
                  f_A : 'a -> ['a - `A],
                  f_B : 'b -> ['b - `B],
                  f_C : 'c -> ['c - `C]) => {
  f_C(f_B(f_A(x)));
};

let refine_test = (refine : 'a -> _ -> _ -> _ -> ['a - `A | `B]);
