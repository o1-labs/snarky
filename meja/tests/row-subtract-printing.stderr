File "tests/row-subtract-printing.meja", line 8, characters 19-25:
Error: Incompatible types
       ([>] as 'b) -> (([>] as 'b) -> [([>] as 'c) - `A])
       -> ([([>] as 'c) - `A] -> [([>] as 'c) - `A | `B])
       -> ([([>] as 'c) - `A | `B] -> [([>] as 'c) - `A | `B | `C])
       -> [([>] as 'b) - `A | `B] and
       ([>] as 'b) -> (([>] as 'b) -> [([>] as 'c) - `A])
       -> ([([>] as 'c) - `A] -> [([>] as 'c) - `A | `B])
       -> ([([>] as 'c) - `A | `B] -> [([>] as 'c) - `A | `B | `C])
       -> [([>] as 'c) - `A | `B | `C]:
       Cannot unify [([>] as 'b) - `A | `B] and [([>] as 'c) - `A | `B | `C]
